frontend/src/components/files/UploadFileDragAndDrop.tsx

```diff
@@ -184,32 +184,37 @@ export const UploadFileDragAndDrop: React.FC<UploadFileDragAndDropProps> = (
 				<Stepper activeStep={activeStep} orientation="vertical">
 					{/*<Stepper activeStep={activeStep}>*/}
 					{/*step 1 Metadata*/}
-					<Step key="fill-in-metadata">
-						<StepLabel>Fill In Metadata</StepLabel>
-						<StepContent TransitionProps={{ unmountOnExit: false }}>
-							<Typography>Provide us the metadata about your file.</Typography>
-							<Box>
-								<CreateMetadata
-									setMetadata={setMetadata}
-									sourceItem={"files"}
-								/>
-							</Box>
-							{/*buttons*/}
-							<Grid container>
-								<Grid xs={11}>
-									<Button
-										variant="contained"
-										onClick={handleNext}
-										disabled={!checkIfFieldsAreRequired() ? false : !allFilled}
-										sx={{ display: "block", marginLeft: "auto" }}
-									>
-										Next
-									</Button>
+					{metadataDefinitionList?.length > 0 && (
```

Comment:
I don't have any file metadata but I do have some dataset metadata. When uploading files it still prompt me to fill metadata
<img width="1258" alt="image" src="https://github.com/clowder-framework/clowder2/assets/13950475/e61ee98b-2cdc-4546-86f5-3b87cc067ced">

I think merely check metadataDefinitionList might not be enough since it doesn't differentiate dataset metadata vs file metadata; Nor does it differentiate "required" vs "not required".

Reply:
We need to decide if we want to consider those. Right now I just checked if there is no metadata defn, it should not show Step 1. If there is any optional metadata defn, users might still want to show them.